ls1intum/Artemis
# `Lectures`: Reimplement attachment file upload

### Checklist
#### General


#### Server
#### Client

### Motivation and Context


Follow-up of #5733 and #5427 to replace the file upload before entity persistence with a multipart file and entity upload.

### Description

I simply extended the endpoints to save an attachment to also allow a file upload (forced for creation), save one REST call and make the system safer. No UI changes were necessary..

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Student

1. Create a lecture in a course
2. Navigate to the "Attachments" of this lecture (not the attachment units!)

- Create an attachment
- Update the attachment with a new file
- Update the attachment without a new file
- Make sure that the student can see the file and the changes correctly

### Review Progress






#### Performance Review
#### Code Review
#### Manual Tests

### Test Coverage



I added assertions where needed. However, for most of the changes, the existing assertions were already enough.
| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|-----------------------------:|
| AttachmentResource.java | 98.1% | ✅ |
| attachment.service.ts | 95.9% | ✅ |
| lecture-attachments.component.ts | 99% | ✅ |
